examples/halo_finder/ArborX_HaloFinder.hpp

```diff
@@ -0,0 +1,476 @@
+struct Wrapped
+{
+  View _M_view;
+  double _r;
+};
+
+template <typename View>
+auto wrap(View v, double r)
+{
+  return Wrapped<View>{v, r};
+}
+
+namespace Traits
+{
+template <typename View>
+struct Access<Wrapped<View>, PredicatesTag>
+{
+  using memory_space = typename View::memory_space;
+  static size_t size(Wrapped<View> const &w) { return w._M_view.extent(0); }
+  static KOKKOS_FUNCTION auto get(Wrapped<View> const &w, size_t i)
+  {
+    return attach(intersects(Sphere{w._M_view(i), w._r}), (int)i);
+  }
+};
+} // namespace Traits
+
+namespace HaloFinder
+{
+
+template <typename ExecutionSpace, typename IndicesView, typename OffsetView,
+          typename CCSView>
+bool verifyCC(ExecutionSpace exec_space, IndicesView indices, OffsetView offset,
+              CCSView ccs)
+{
+  int num_nodes = ccs.size();
+  ARBORX_ASSERT((int)offset.size() == num_nodes + 1);
+  ARBORX_ASSERT(ArborX::lastElement(offset) == (int)indices.size());
+
+  // Check that connected vertices have the same cc index
+  int num_incorrect = 0;
+  Kokkos::parallel_reduce(
+      ARBORX_MARK_REGION("verify_connected_indices"),
+      Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_nodes),
+      KOKKOS_LAMBDA(int i, int &update) {
+        for (int j = offset(i); j < offset(i + 1); ++j)
+        {
+          if (ccs(i) != ccs(indices(j)))
+          {
+            printf("Non-matching cc indices: %d [%d] -> %d [%d]\n", i, ccs(i),
+                   indices(j), ccs(indices(j)));
+            update++;
+          }
+        }
+      },
+      num_incorrect);
+  if (num_incorrect)
+    return false;
+
+  // Check that non-connected vertices have different cc indices
+  auto ccs_host = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, ccs);
+  auto offset_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, offset);
+  auto indices_host =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, indices);
+
+  std::set<int> unique_cc_indices;
+  for (int i = 0; i < num_nodes; i++)
+    unique_cc_indices.insert(ccs_host(i));
+  auto num_unique_cc_indices = unique_cc_indices.size();
+
+  unsigned int num_ccs = 0;
+  std::set<int> cc_sets;
+  for (int i = 0; i < num_nodes; i++)
+  {
+    if (ccs_host(i) >= 0)
+    {
+      auto id = ccs_host(i);
+      cc_sets.insert(id);
+      num_ccs++;
+
+      // DFS search
+      std::stack<int> stack;
+      stack.push(i);
+      while (!stack.empty())
+      {
+        auto k = stack.top();
+        stack.pop();
+        if (ccs_host(k) >= 0)
+        {
+          ARBORX_ASSERT(ccs_host(k) == id);
+          ccs_host(k) = -1;
+          for (int j = offset_host(k); j < offset_host(k + 1); j++)
+            stack.push(indices_host(j));
+        }
+      }
+    }
+  }
+  if (cc_sets.size() != num_unique_cc_indices)
+  {
+    // FIXME: Not sure how we can get here, but it was in the original verify
+    // check in ECL
+    std::cout << "Number of components does not match" << std::endl;
```

Comment:
```suggestion
    std::cerr << "Number of components does not match\n";
```

Reply:
You meant to keep the flush?

Reply:
I didn't notice. But I think it's not worth changing it.